src/components/inputs/date-input/date-input.js

```diff
@@ -25,267 +25,237 @@ import {
   parseInputToDate,
 } from '../../../utils/calendar';
 
-class DateInput extends React.Component {
-  static displayName = 'DateInput';
+const DateInput = props => {
+  const intl = useIntl();
+  const [calendarDate, setCalendarDate] = React.useState(
+    props.value || getToday()
+  );
+  const [suggestedItems, setSuggestedItems] = React.useState([]);
+  const [highlightedIndex, setHighlightedIndex] = React.useState(
+    props.value === '' ? null : getDateInMonth(props.value) - 1
+  );
+  const inputRef = React.useRef();
 
-  static isEmpty = value => value === '';
-
-  static propTypes = {
-    intl: PropTypes.shape({
-      locale: PropTypes.string.isRequired,
-      formatMessage: PropTypes.func.isRequired,
-    }).isRequired,
-    horizontalConstraint: PropTypes.oneOf(['m', 'l', 'xl', 'scale']),
-    value: PropTypes.string.isRequired,
-    onChange: PropTypes.func.isRequired,
-    onFocus: PropTypes.func,
-    onBlur: PropTypes.func,
-    id: PropTypes.string,
-    name: PropTypes.string,
-    placeholder: PropTypes.string,
-    isDisabled: PropTypes.bool,
-    hasError: PropTypes.bool,
-    hasWarning: PropTypes.bool,
-  };
-
-  inputRef = React.createRef();
-
-  state = {
-    calendarDate: this.props.value || getToday(),
-    suggestedItems: [],
-    highlightedIndex:
-      this.props.value === '' ? null : getDateInMonth(this.props.value) - 1,
-  };
-
-  showPrevMonth = () => {
-    this.setState(prevState => ({
-      calendarDate: changeMonth(prevState.calendarDate, -1),
-      // select first day in next month
-      highlightedIndex: prevState.suggestedItems.length,
-    }));
-  };
-
-  jumpMonth = amount => {
-    this.setState(prevState => {
-      const nextDate = changeMonth(prevState.calendarDate, amount);
-      return { calendarDate: nextDate, highlightedIndex: 0 };
-    });
-  };
-
-  showToday = () => {
-    const today = getToday();
-    this.setState(
-      prevState => ({
-        calendarDate: today,
-        highlightedIndex:
-          prevState.suggestedItems.length + getDateInMonth(today) - 1,
+  const emit = React.useCallback(
+    value =>
+      props.onChange({
+        target: {
+          id: props.id,
+          name: props.name,
+          // when cleared the value is null, but we always want it to be an
+          // empty string when there is no value.
+          value: value || '',
+        },
       }),
-      () => this.inputRef.current.focus()
-    );
-  };
+    [props.id, props.name]
+  );
+
+  const handleChange = React.useCallback(
+    date => {
+      inputRef.current.setSelectionRange(0, 100);
+      emit(date);
+    },
+    [inputRef]
```

Comment:
I'm not sure if this makes sense 🤔

Reply:
To my understanding yes. useCallback does not have smart exceptions like any value of props or any ref. It is just a closure and memoization as far as I understood it.